MDANSE/Src/MDANSE/Framework/Jobs/TrajectoryFilter.py

```diff
@@ -0,0 +1,232 @@
+#    This file is part of MDANSE.
+#
+#    MDANSE is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+import collections
+import copy
+import json
+
+import numpy as np
+
+import h5py
+from MDANSE.Framework.Formats import HDFFormat
+
+from MDANSE.Chemistry.ChemicalEntity import AtomGroup
+from MDANSE.Framework.Jobs.IJob import IJob
+from MDANSE.Mathematics.Signal import filter_map
+from MDANSE.MolecularDynamics.Configuration import RealConfiguration
+from MDANSE.MolecularDynamics.Trajectory import sorted_atoms, TrajectoryWriter
+from MDANSE.MLogging import LOG
+
+
+class TrajectoryFilter(IJob):
+    """
+    Design and apply a filter for the atomic trajectories.
+    """
+
+    label = "Trajectory Filter"
+
+    category = (
+        "Analysis",
+        "Dynamics",
+    )
+
+    ancestor = ["hdf_trajectory", "molecular_viewer"]
+
+    settings = collections.OrderedDict()
+    settings["trajectory"] = ("HDFTrajectoryConfigurator", {})
+    settings["frames"] = (
+        "CorrelationFramesConfigurator",
+        {"dependencies": {"trajectory": "trajectory"}},
+    )
+    settings["instrument_resolution"] = (
+        "InstrumentResolutionConfigurator",
+        {"dependencies": {"trajectory": "trajectory", "frames": "frames"}},
+    )
+    settings["projection"] = (
+        "ProjectionConfigurator",
+        {"label": "project coordinates"},
+    )
+    settings["trajectory_filter"] = (
+        "TrajectoryFilterConfigurator",
+        {"dependencies": {"trajectory": "trajectory"}},
+    )
+    settings["atom_selection"] = (
+        "AtomSelectionConfigurator",
+        {"dependencies": {"trajectory": "trajectory"}},
+    )
+    settings["atom_transmutation"] = (
+        "AtomTransmutationConfigurator",
+        {
+            "dependencies": {
+                "trajectory": "trajectory",
+                "atom_selection": "atom_selection",
+            }
+        },
+    )
+    settings["weights"] = (
+        "WeightsConfigurator",
+        {
+            "default": "atomic_weight",
+            "dependencies": {"atom_selection": "atom_selection"},
+        },
+    )
+    settings["output_files"] = (
+        "OutputTrajectoryConfigurator",
+        {"format": "MDTFormat"},
+    )
+    settings["running_mode"] = ("RunningModeConfigurator", {})
```

Comment:
Some of these settings are not used and/or are not relevant to the filter job. They can be removed, this will clear up the actions tab as well.